cmake/CMakeLists.txt

```diff
@@ -25,15 +25,28 @@ endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CXX_FLAGS)
 
 file(GLOB SRC_FILES ${LAMMPS_SOURCE_DIR}/*.cpp)
 list(SORT SRC_FILES)
-# check for files installed by make-based buildsystem 
-# only run this time consuming check if there are new files
+
+# check for files auto-generated by make-based buildsystem
+# this is fast, so check for it all the time
+message(STATUS "Running check for auto-generated files from make-based build system")
+file(GLOB SRC_AUTOGEN_FILES ${LAMMPS_SOURCE_DIR}/style_*.h)
+list(APPEND SRC_AUTOGEN_FILES ${LAMMPS_SOURCE_DIR}/lmpinstalledpkgs.h)
+foreach(_SRC ${SRC_AUTOGEN_FILES})
```

Comment:
Even if it is fast, I think we should still cache `SRC_AUTOGEN_FILES` as well.

Reply:
there really is nothing to cache here. none of these files should exist after a successful cmake run.

Reply:
Good point!

Reply:
@junghans yes there is logic to ignore any `*.h` and `*.cpp` file which was copied into the `src/` directory but actually comes from one of the packages (see  #628) and https://github.com/lammps/lammps/blob/master/cmake/CMakeLists.txt#L473. If I remember correctly that takes care of the case that Steve likes to already install packages in the tarball releases, but that would break the cmake build. The consensus at that time was to ignore files in the `src` directory that come from a package. So in principle, https://github.com/lammps/lammps/blob/master/cmake/CMakeLists.txt#L30 shouldn't be necessary and we only have to change the "silent" behaviour and move the error message.  However, these changes from Axel are about the style files still being present.